### ISSUE TITLE
Get `Reason` from `GoogleError`

### ISSUE THREAD

**Issue by ralpha** [ThouCheese/cloud-storage-rs#76]
The api does not expose the `reason`, `message` or other data for [`GoogleError`](https://docs.rs/cloud-storage/0.10.2/cloud_storage/struct.GoogleError.html) directly.

I can not do the following:
```rust
let error: GoogleError = { /* error from somewhere */ };
match error.reason {
    BadRequest => { println!("Some message"); }
    ParseError => { println!("Some other message"); }
    _ => { println!("Other errors"); }
}
```

Also `GoogleError` does not implement `Display` so can not just use it in a `format!` or `println!` statement. (only `Debug` is implemented)

**Comment by ThouCheese** [comment]
You can use `GoogleError` in format or println by using the debug formatter right? Just write `println!("{:?}", e)` and you should be good. Nonetheless I have implemented Display and made the fields public, please let me know if #79 is what you had in mind.

**Comment by ralpha** [comment]
Yes that looks great. 
Now I can loop over the errors and do a `match` on the reason and handle them separately.
This makes the error message also a bit nicer.

**Comment by ThouCheese** [comment]
Fixed in `0.10.3`
